packages/marshal/src/dot-membrane.js

```diff
@@ -0,0 +1,152 @@
+/* eslint-disable no-use-before-define */
+// @ts-check
+
+// eslint-disable-next-line spaced-comment
+/// <reference types="ses"/>
+
+import { assert, details as X } from '@agoric/assert';
+import { E } from '@agoric/eventual-send';
+import { isObject } from './helpers/passStyleHelpers.js';
+import { getInterfaceOf } from './helpers/remotable.js';
+import { Far } from './make-far.js';
+import { makeMarshal } from './marshal.js';
+import { passStyleOf } from './passStyleOf.js';
+
+const { fromEntries } = Object;
+const { ownKeys } = Reflect;
+
+const makeConverter = (mirrorConverter = undefined) => {
+  /** @type {WeakMap<any,any>=} */
+  let mineToYours = new WeakMap();
+  let optReasonString;
+  const myRevoke = reasonString => {
+    assert.typeof(reasonString, 'string');
+    mineToYours = undefined;
+    optReasonString = reasonString;
+    if (optInnerRevoke) {
+      optInnerRevoke(reasonString);
+    }
+  };
+  const convertMineToYours = (mine, _optIface = undefined) => {
+    if (mineToYours === undefined) {
+      throw harden(ReferenceError(`Revoked: ${optReasonString}`));
+    }
+    if (mineToYours.has(mine)) {
+      return mineToYours.get(mine);
+    }
+    let yours;
+    const passStyle = passStyleOf(mine);
+    switch (passStyle) {
+      case 'promise': {
+        let yourResolve;
+        let yourReject;
+        yours = new Promise((res, rej) => {
+          yourResolve = res;
+          yourReject = rej;
+        });
+        E.when(
+          mine,
+          myFulfillment => {
+            yourResolve(pass(myFulfillment));
+          },
+          myReason => {
+            yourReject(pass(myReason));
+          },
+        ).catch(metaReason => {
+          // This can happen if myFulfillment or myReason is not passable.
+          // TODO verify that metaReason must be your-side-safe
+          yourReject(metaReason);
+        });
```

Comment:
I'd like to give a best-effort attempt to pass the metaReason, and handle when the catch block could still throw.

```suggestion
          yourReject(pass(metaReason));
        }).catch(metaMetaReason => yourReject(metaMetaReason));
```

Reply:
It is hard to tell what the right thing is here, but I like your's better than mine. Done. Resolving.